testbed/tests/app/test_desktop.py

```diff
@@ -168,159 +170,183 @@ async def test_menu_minimize(app, app_probe):
-    """Window can be made full screen"""
-    window1 = toga.Window("Test Window 1", position=(150, 150), size=(200, 200))
-    window2 = toga.Window("Test Window 2", position=(400, 150), size=(200, 200))
-
-    window1_widget = toga.Box(style=Pack(flex=1))
-    window2_widget = toga.Box(style=Pack(flex=1))
-    window1_widget_probe = get_probe(window1_widget)
-    window2_widget_probe = get_probe(window2_widget)
-
-    window1.content = toga.Box(
-        children=[window1_widget], style=Pack(background_color=REBECCAPURPLE)
+async def test_presentation_mode(app, app_probe, main_window, main_window_probe):
+    """The app can enter presentation mode."""
+    window_information_list = list()
+    windows_list = list()
+    for i in range(len(app.screens)):
+        window = toga.Window(title=f"Test Window {i}", size=(200, 200))
+        r = random.randint(0, 255)
+        g = random.randint(0, 255)
+        b = random.randint(0, 255)
+        window_widget = toga.Box(
+            style=Pack(flex=1, background_color=f"#{r:02X}{g:02X}{b:02X}")
+        )
+        window.content = window_widget
+        window.show()
+
+        window_information = dict()
+        window_information["window"] = window
+        window_information["window_probe"] = window_probe(app, window)
+        window_information["initial_content_size"] = window_information[
+            "window_probe"
+        ].content_size
+        window_information["widget_probe"] = get_probe(window_widget)
+        window_information["initial_widget_size"] = (
+            window_information["widget_probe"].width,
+            window_information["widget_probe"].height,
+        )
+        window_information_list.append(window_information)
+        windows_list.append(window)
+
+    # Add delay to ensure windows are visible after animation.
+    await main_window_probe.wait_for_window("All Test Windows are visible")
+
+    screen_window_dict = dict()
+    for window, screen in zip(windows_list, app.screens):
+        screen_window_dict[screen] = window
+
+    # Enter presentation mode with a screen-window dict via the app
+    app.enter_presentation_mode(screen_window_dict)
+    # Add delay to ensure windows are visible after animation.
+    await main_window_probe.wait_for_window(
+        "App is in presentation mode", full_screen=True
     )
-    window2.content = toga.Box(
-        children=[window2_widget], style=Pack(background_color=CORNFLOWERBLUE)
+    assert app.in_presentation_mode
+    # All the windows should be in presentation mode.
+    for window_information in window_information_list:
+        assert (
+            window_information["window"].state == WindowState.PRESENTATION
+        ), f"{window_information['window'].title}:"
+        assert (
+            window_information["window_probe"].content_size[0] > 1000
+        ), f"{window_information['window'].title}:"
+        assert (
+            window_information["window_probe"].content_size[1] > 700
```

Comment:
Why 1000x700? Magic numbers should always be explained in comments.

Reply:
I am not sure, the previous tests had these specific comparison values, so I had kept them as is.

Reply:
Ok - then this is "do as I say, not as I do 😝". 

My guess is that 1000x700 is a size that is clearly bigger that the original window size, while being smaller than any likely screen.

Reply:
I have added a comment.